### PR TITLE
Allow egrees from istio-ingressgateway pod if APIServerSNI feature gate is enabled

### LINKED ISSUE #6036: Creation of a `Shoot` cluster can fail, if it is the first `Shoot` assigned to a `Seed`
**How to categorize this issue?**

/area usability
/kind bug

**What happened**:
When the `APIServerSNI` feature gate is enabled, the sync period of the seed controller in `gardenlet` is relatively large and the dependency watchdog probe is enabled (`seed.spec.settings.dependencyWatchdog.probe.enabled=true`), creation of a shoot cluster (if that is the first shoot cluster on the seed) will fail initially because the `dependency-watchdog`'s external kube-apiserver probe fails and therefore KCM, MCM and CA are all scaled down. The probe will become successful after the seed has been reconciled and only if there already exists an istio `virtualservice` for the shoot's kube-apiserver.

All of this happens because during seed bootstrap the following code is used to detect whether SNI is enabled or not:
https://github.com/gardener/gardener/blob/8f819ffab82898c65be19a4376aeec3b5627c3ba/pkg/operation/seed/seed.go#L738
https://github.com/gardener/gardener/blob/8f819ffab82898c65be19a4376aeec3b5627c3ba/pkg/operation/botanist/component/kubeapiserverexposure/kube_apiserver_sni.go#L224-L237
And then a specific rule is added to the `allow-to-all-shoot-apiservers` `Networkpolicy` depending on whether SNI is enabled:
https://github.com/gardener/gardener/blob/e7a25ef16f7452a7b81cf1a2c0724ab41d4af886/pkg/operation/botanist/component/networkpolicies/global.go#L162-L172

So if a `virtualservice` is not yet created for any shoots, `anySNI` will be evaluated to false. This will result in the `allow-to-all-shoot-apiservers` `Networkpolicy`  not containing a rule that allows access to the `istio-ingressgateway` pod and therefore the `dependency-watchdog` cannot talk to it when trying to evaluate the external `kube-apiserver` probe. 
When `gardenlet`'s seed controller sync period is relatively small (~1m) this issue is not noticeable as the seed will be bootstrapped and `anySNY` evaluated every minute. Then the `allow-to-all-shoot-apiservers` `Networkpolicy` will be adapted accordingly before `dependency-watchdog`'s external `kube-apiserver` probe has a chance to fail.

**What you expected to happen**:
The `allow-to-all-shoot-apiservers` `Networkpolicy` to contain the appropriate rule:
```
- namespaceSelector: {}
  podSelector:
    matchLabels:
    app: istio-ingressgateway
```
 independent of `gardenlet`'s seed controller sync period when the `APIServerSNI` feature gate is enabled.

**How to reproduce it (as minimally and precisely as possible)**:
In `gardenlet`'s component config set the `controllers.seed.syncPeriod` to 1h. Enable the `APIServerSNI` feature gate. Create a seed cluster and make sure that `seed.spec.settings.dependencyWatchdog.probe.enabled` is set to true and that there are no shoots already running on the seed. In the seed's `garden` namespace you can see that the `allow-to-all-shoot-apiservers` `Networkpolicy` does not contain the aforementioned rule. 
Create a shoot cluster and observe that it's KCM, MCM and CA all get scaled down. Once the 1h of the sync period has elapsed (or if you simply restart `gardenlet`), notice that `allow-to-all-shoot-apiservers` is edited accordingly and KCM, MCM and CA all get scaled up.

**Anything else we need to know?**:
Discovered this as part of: https://github.com/gardener/gardener/pull/5995#issuecomment-1138921606

**Environment**:

- Gardener version: e7a25ef16f7452a7b81cf1a2c0724ab41d4af886

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area usability
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #6036

**Special notes for your reviewer**:
/cc @plkokanov 

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: charts/gardener/gardenlet/values.yaml [hunk 1]
```diff
@@ -79,9 +79,7 @@ global:
         # - production
         seed:
           concurrentSyncs: 5
-        # TODO (plkokanov): the sync period is currently set to 1m because of the way that SNI detection is done during seed reconciliations.
-        # ref: https://github.com/gardener/gardener/issues/6036
-          syncPeriod: 1m
+          syncPeriod: 15m
```
Comment (ary1992) [inline charts/gardener/gardenlet/values.yaml]: How you came up with the value of `syncPeriod` as `15m` ?

**Review by ary1992 — COMMENTED** [review]
Just a small question

File: charts/gardener/gardenlet/values.yaml (hunk 1 above)
Comment (acumino) [reply charts/gardener/gardenlet/values.yaml]: I haven't deduced it, I took the value from this [PR](https://github.com/gardener/gardener/pull/5995).

Comment (plkokanov) [reply charts/gardener/gardenlet/values.yaml]: We would have used `1h` in the PR that you linked: https://github.com/gardener/gardener/pull/5995#discussion_r879086695

File: pkg/operation/seed/seed.go [hunk 2]
```diff
@@ -739,6 +739,7 @@ func RunReconcileSeedFlow(
 	if err != nil {
 		return err
 	}
+	anySNI = anySNI || gardenletfeatures.FeatureGate.Enabled(features.APIServerSNI)
```
Comment (plkokanov) [inline pkg/operation/seed/seed.go]: Probably it is better to rename this to
```suggestion
	sniEnabledOrInUse = anySNI || gardenletfeatures.FeatureGate.Enabled(features.APIServerSNI)
```

File: pkg/operation/seed/seed.go [hunk 3]
```diff
@@ -776,7 +777,7 @@ func RunReconcileSeedFlow(
 
 		// even if SNI is being disabled, the existing ports must stay the same
 		// until all APIServer SNI resources are removed.
-		if gardenletfeatures.FeatureGate.Enabled(features.APIServerSNI) || anySNI {
+		if anySNI {
```
Comment (plkokanov) [inline pkg/operation/seed/seed.go]:
```suggestion
		if sniEnabledOrInUse {
```

**Review by plkokanov — CHANGES_REQUESTED** [review]
Just some minor stuff that we found while looking into the PR

**Comment by rfranzke** [comment]
@acumino Could you please address the nit comments so that we can get this PR merged? Thanks! 🤝

**Pull request merged**